File: src/main/java/io/awilson/timein/services/CourseServiceImpl.java
package io.awilson.timein.services;

import io.awilson.timein.domain.Course;
import io.awilson.timein.repositories.CourseRepository;
import lombok.Setter;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class CourseServiceImpl implements CourseService {

    @Setter(onMethod = @_(@Autowired))
    private CourseRepository courseRepository;

    @Override
    public Iterable<Course> listAllCourses() { return courseRepository.findAll(); }

    @Override
    public Course getCourseById(Integer id) {
        return courseRepository.findOne(id);
    }

    @Override
    public Course saveCourse(Course course) {
        return courseRepository.save(course);
    }

    @Override
    public void deleteCourse(Integer id) {
        courseRepository.delete(id);
    }
}

File: src/main/java/io/awilson/timein/services/CourseServiceImpl.kt
package io.awilson.timein.services

import io.awilson.timein.domain.Course
import io.awilson.timein.repositories.CourseRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class CourseServiceImpl : CourseService {

    @Autowired
    lateinit var repo: CourseRepository

    override fun listAll(): Iterable<Course> = repo.findAll()

    override fun getById(id: Int): Course = repo.findOne(id) ?: throw Exception("Course:$id not found")

    override fun save(course: Course): Course = repo.save(course)

    override fun delete(id: Int) = repo.delete(id)
}